ludwig/features/category_feature.py

```diff
@@ -187,11 +188,22 @@ def predictions(
         )
         predictions = tf.cast(predictions, dtype=tf.int64)
 
-        return {
-            PREDICTIONS: predictions,
-            PROBABILITIES: probabilities,
-            LOGITS: logits
-        }
+        if self.loss[TYPE] == 'sampled_softmax_cross_entropy':
+            return {
+                PREDICTIONS: predictions,
+                PROBABILITIES: probabilities,
+                LOGITS: logits,
+
+                # required for evaluation loss computation
+                LAST_HIDDEN: inputs[LAST_HIDDEN]
```

Comment:
I may be wrong, but shouldn-t this be `PROJECTION_INPUT: inputs[LAST_HIDDEN]`?

Reply:
I think you may be correct, this would be consistent with what we discussed re: sequences.  the issue is `LAST_HIDDEN` is referenced elsewhere in ECD, loss_modules and predictor modules.  So I have to be a bit careful in changing this.  I just need to confirm the usage is only in the context of categorical or sequence features.

Reply:
maybe replacing all `PROJECTION_INPUT` to `LAST_HIDDEN` may be a clean solution, if there are no conflicts.

Reply:
@w4nderlust   The line (198) cited is in the categorical feature's prediction method.  As I learned in the sequence sampled softmax loss work, the sampled softmax loss only applies during training.  This means that lines 197-198 are not needed because I should be using the regular softmax.   Lines 197 and 198 are present because the work pre-date working sampled softmax for sequences.  So I need to re-work the categorical feature to reflect update view of using sampled softmax loss calculation.

I believe the correct action is to remove Lines 197 and 198.  However,  I still need to find the right place in the categorical training process to insert the `PROJECTION_INPUT` key.    Another reason for putting in the `PROJECTION_INPUT` key is that the key `LAST_HIDDEN` is involved in handling output feature dependencies function.   So it would be good to not overload the `LAST_HIDDEN` key for two different purposes: sampled softmax loss and keeping track of dependencies.

Reply:
I resolved this comment as follows:
* removed the cited lines because these are not needed during prediction.  The removed tensor is needed only during training with sampled softmax.
* Reworked the categorical output `_setup_loss()` and `_setup_metric()` methods to ensure sampled softmax function is only used during training.
* Updated categorical output feature `logits()` method to expose the `PROJECTION_INPUT tensor required for sampled softmax.
* Slight refactoring of `OutputFeature.call()` method to handle the `PROJECTION_INPUT` tensor from the categorical output feature.

Reply:
this looks fine to me!